### ISSUE TITLE
🔑 sdd

### ISSUE THREAD

**Issue by WallKey1** [SoftbearStudios/mk48#162]
**Describe the bug**

**Expected behavior**

**Additional context (optional)**

**Comment by WallKey1** [comment]
Why is it so hard to distinguish between some land/compact ice and ice I can travel under. 3 or 4 areas look passable but are not.

**Comment by WallKey1** [comment]
1205e, 3239n 538e 3248n

**Comment by WallKey1** [comment]
Have all graphics maxed out. Several more like this.

**Comment by WallKey1** [comment]
1074e 2561n

**Comment by finnbear** [comment]
![image](https://user-images.githubusercontent.com/20015102/155820135-f325bfca-835e-47c1-9ca1-922013bc18d6.png)
For reference, the middle of this image is deadly snow.

**Comment by WallKey1** [comment]
Thanks - My team managed to get iulik to 115,764 today without going under the ice death traps. Will try to post video, may help other people get better at the game. As for now my goal was to knock toad off the leader board and we did it! May return to play the game if someone beats this score ;)

**Comment by finnbear** [comment]
> My team managed to get iulik to 115,764 today 

I'm speechless, and curious! How long did it take? How large was your team? What was your strategy? (if the video answers these questions, I'd gladly watch it)

Anyway, thanks for playing!

**Comment by WallKey1** [comment]
Took about 8 hours, I spent 3/4 building a wall with a couple dredgers. Then a failed attempt where I got betrayed at 37K but Iulik survived with about 25k. So we decided to focus on getting Iulik to 60k then 60k moved to the goal of 100k.  I managed to get to 32k again and donate all this to iulik. But keeping iulik alive & gathering xp to donate was the main strategy. 

Then to 200k but due to time zones etc people were leaving the team. And  a well known player joined the server and team upgrade killed Iulik by beaching.

Also having at least 2 players constantly repairing holes in the wall. I think 200k would be possible if I could get 4 good players to stay on for 8 hours but extremely hard to achieve this. 

Timing is also important, playing when Antifa, doggo, & aaaaaa are asleep/away is also "key", as they will happily commit suicide to kill anyone.

**Comment by caibear** [comment]
A mitigation for this will be released in the next update. Terrain will now repel boats that collide with it. This will hopefully give players time to reverse or turn away from the land.

**Comment by mitm001** [comment]
I have another solution. Make land repel sea ice so we can identify it better. This would give it a visible border. You could dirty up the sea ice some to help. 

Repelling is not really a fix because subs would know if there is "deadly snow" and never collide with it. It would be just as obvious as land.

**Comment by finnbear** [comment]
> I have another solution. Make land repel sea ice so we can identify it better. This would give it a visible border. You could dirty up the sea ice some to help.

The issue here relates to the root-cause of this problem. It's not that the shader intentionally makes the difference between snow and ice too hard to see. The problem is that the shader doesn't have precise knowledge of where ice ends and land begins due to systematic differences between the terrain interpolation algorithm on the server's CPU and the terrain interpolation pipeline on the client's GPU. In rare cases, these two process disagree on where the deadly snow is, and as of now, we have not found a way to avoid this. The repulsion solution helps by allowing the server to communicate its perception of where snow is (the ground truth) in a less-deadly manner.

**Comment by mitm001** [comment]
Your solution hasn't gone live yet right?

**Comment by finnbear** [comment]
> Your solution hasn't gone live yet right?

Correct. It will go live *next* update, for which there is no precise ETA. We usually update when the benefits far outweigh the cost of switching servers, which usually means we end up making many changes per update instead of releasing every change in its own update.

**Comment by mitm001** [comment]
Whats the cost involved? 

I am up to 15 deaths now. Sea Wolfs last half dozen or more. This is a significant bug fix. This last time I was barely moving. I even cleared 3 hidden pieces with torpedoes prior to this one that sank me as I was barely able to pull out of them in time. All grouped around each other and identical to surface ice.

That is 4 pieces grouped but not obviously land. I even fired a torpedo in front of me to see if the path was clear but still hit one. First time I have seen (well didn't see but you get the drift) that many grouped together.

Worth a restart imo.

**Comment by mitm001** [comment]
I am unsure if you can determine ice land from others, but if you can, would increasing the height of the ice land help the shader distinguish the difference easier?

**Comment by WallKey1** [comment]
O_O

**Comment by WallKey1** [comment]
Please make more space on map, I like to hide. 

Tanti Saluti,
K 😘

**Comment by finnbear** [comment]
> Whats the cost involved?

The cost of switching servers is as follows:
- Players on the old server, most of whom don't understand what is happening, will gradually be exposed to an empty/dead-looking server
- The new server, for the first ~30 minutes of its existence, also looks relatively empty/dead.
- We monitor the new server for an hour to make sure it is running smoothly
- Around 10% of updates cause some issue that we have to immediately patch, causing stress on our part and doubling the costs.
- At least 3 players ask if there are any new ships, to which the answer would be very disappointing to everyone reading chat

**Comment by WallKey1** [comment]
Something needs to change if serious on scaling this, it's got stagnant recently. If that's not the vision for the project that's cool too. I like to check it out every few weeks kinda interesting to follow.

**Comment by WallKey1** [comment]
I slag the game off too much sometimes & forget it's someone's baby. Sorry

**Comment by finnbear** [comment]
> I slag the game off too much sometimes & forget it's someone's baby. Sorry

No problem. This issue remains open because it remains valid. If it wasn't a valid issue, we would have closed it and/or marked it "wontfix."

As previously mentioned, the next update (for which there is no ETA) will add a repulsive force when you contact land from the server's perspective. This is intended to mitigate the issue.

**Comment by finnbear** [comment]
> The map when over 100 people are online is so much better

The map expands to an area proportional to the number of players. This expansion causes more arctic area to exist, as the latitude of arctic is constant but the upper border expands.

> is the counter still broken also or has it been fixed?

The player counter is accurate, if that's what you mean.

> Would be nice if my team doesn't keep getting attacked by the same person/s who "magically" appears out of nowhere when we reach higher scores recently. O_O

Unless they are spawning next to you, there isn't any magic in the game ;)

> What's the ETA for LV10 ?

No ETA.

> I appreciate they may just be a side hustle and things take time etc....

It's not just a question of the work taking time. I'm just really busy with school work, and when I do have free time, I'm trying to develop more games. More Mk48.io fixes, ships, and features are planned for when we do have time, though.

> Would love to help but can't programme Rust :eye: :lips: :eye:

You could learn it (it's a great language, so you won't regret it)!

**Comment by WallKey1** [comment]
Yeah the same person suspiciously appears O_O 

If the map was bigger by default that would make the game more interesting, there is little ICE when 40/50 players are online.

**Comment by mitm001** [comment]
The latest fix is helpful but not quite complete as things are still the same if traveling in reverse, which you do when defending yourself.
